.github/workflows/python.yaml

```diff
@@ -72,4 +72,11 @@ jobs:
       with:
         version: ${{ env.CURRENT_VERSION }}
         alias: development
-        push: ${{ github.ref == 'refs/heads/main' }}  # build always, publish on 'main' only to prevent version clutter
+        push: ${{ github.ref == 'refs/heads/main' }} # build always, publish on 'main' only to prevent version clutter
+    - name: Execute jupytext in demo venv
+      run: |
+        python -m venv .demovenv
+        source .demovenv/bin/activate
+        pip install -r demos/requirements.txt
```

Comment:
I support the idea of creating a separate venv for the docs generation sounds sensible.

If we stick with mkdocs for the docs generation (I would like to include the examples in the docs) for uniformity and simplicity, we'd also have to install all dependencies from `requirements-docs.txt` in this venv.